pkg/sqlite/directory.go

```diff
@@ -20,14 +20,68 @@ const ClusterServiceVersionKind = "ClusterServiceVersion"
 type SQLPopulator interface {
 	Populate() error
 }
+type SQLRemover interface {
+	Remove() error
+}
 
 // DirectoryLoader loads a directory of resources into the database
 type DirectoryLoader struct {
 	store     registry.Load
 	directory string
 }
 
+// PackageRemover loads a directory of resources into the database
+type PackageRemover struct {
+	store    registry.Load
+	packages string
+}
+
 var _ SQLPopulator = &DirectoryLoader{}
+var _ SQLRemover = &PackageRemover{}
+
+func NewSQLRemoverForPackages(store registry.Load, packages string) *PackageRemover {
+	return &PackageRemover{
+		store:    store,
+		packages: packages,
+	}
+}
+
+func (d *PackageRemover) Remove() error {
+	log := logrus.WithField("pkg", d.packages)
+
+	log.Info("deleting packages")
+
+	var errs []error
+	packages := sanitizePackageList(strings.Split(d.packages, ","))
+	log.Info("input has been sanitized")
+	log.Infof("packages: %s", packages)
+
+	for _, pkg := range packages {
+		if err := d.store.RmPackageName(pkg); err != nil {
+			errs = append(errs, fmt.Errorf("error removing operator package %s: %s", pkg, err))
+		}
+	}
+
+	return utilerrors.NewAggregate(errs)
+}
+
+// sanitizePackageList sanitizes the set of package(s) specified. It removes
+// duplicates and ignores empty string.
+func sanitizePackageList(in []string) []string {
+	out := make([]string, 0)
+
+	inMap := map[string]bool{}
```

Comment:
nit:

you can just use `map[string]struct{}` for a "hash list" implementation
instead of
`inMap[item] = true`

just use
`inMap[item] = struct{}{}`

In the general use case for that data structure it's nice because you can iterate over the list and get a sort of `is in map` syntax a little more easily and it's slightly less data to pass around.

second nit:
Also what is the purpose of checking `item == ""` in the map check function? Filtering empty string from the list seems like a totally unrelated check. Can we break them apart? It's a little easier to read that way.
```
if item == "" {
	continue
}

if _, ok := inMap[item]; ok {
	continue
}
```

Reply:
this is code I stole from somewhere else in the code base... But yes I agree and happy to fix it